src/inmanta/module.py

```diff
@@ -2778,6 +2778,7 @@ def __init__(self, project: Optional[Project], path: str) -> None:
         self._ast_cache: dict[str, tuple[list[Statement], BasicBlock]] = {}  # Cache for expensive method calls
         self._import_cache: dict[str, list[DefineImport]] = {}  # Cache for expensive method calls
         self._dir_cache: Dict[str, list[str]] = {}  # Cache containing all the filepaths present in a dir
+        self._plugin_file_cache: Optional[list[tuple[Path, ModuleName]]] = None
```

Comment:
this may not be strictly required

Reply:
I'm failing to understand the lifecycle for this cache:
- At project creation all modules will get an empty cache
- after a compile this cache will be set
- during an export it will be read

But once it is set, it's never updated ? Won't we miss some things if for example we 
1 compile to set the cache
2 add some plugins code with dependencies
3 call project freeze to read the cache -> these newly added deps will be missing ?

Reply:
no code can be added during the compile (i.e. the compiler starts fresh every time, so during the compile, is literally the time it takes to run the compile command)